packages/server/jiraImagesHandler.ts

```diff
@@ -1,20 +1,25 @@
 import {promises as fsp} from 'fs'
-import mime from 'mime-types'
 import path from 'path'
 import {HttpRequest, HttpResponse} from 'uWebSockets.js'
 import jiraPlaceholder from '../../static/images/illustrations/imageNotFound.png'
 import sleep from '../client/utils/sleep'
 import uWSAsyncHandler from './graphql/uWSAsyncHandler'
 import getRedis from './utils/getRedis'
 
-const getImageFromCache = async (fileName: string, tryAgain: boolean) => {
+const getImageFromCache = async (imgUrlHash: string, tryAgain: boolean) => {
   const redis = getRedis()
-  const imageBuffer = await redis.getBuffer(fileName)
+  const [[imageErr, imageBuffer], [contentTypeErr, contentType]] = await redis
+    .multi()
+    .getBuffer(`jira-image:${imgUrlHash}`)
+    .get(`jira-image:mime-type:${imgUrlHash}`)
+    .exec()
+
+  if (imageErr || contentTypeErr) return null
   if (imageBuffer === null || imageBuffer.length === 0) return null
-  if (imageBuffer.length > 1) return imageBuffer
+  if (imageBuffer.length > 1) return {imageBuffer, contentType}
```

Comment:
can we trust that contentType will always be non-null?

Reply:
Technically, web servers have to provide a proper content type. Here, I'm just returning whatever was cached in its original shape. Before sending back the response, we're checking both `imageBuffer` and `contentType` so we should be good!

```
  const cachedImage = await getImageFromCache(imgUrlHash, true)
  if (!cachedImage?.imageBuffer || !cachedImage?.contentType) {
    await servePlaceholderImage(res)
    return
  }
```

Reply:
ah good call!